File: 2018/03/3_1.rb
lines = File.readlines("input.txt")

grid = Hash.new(0)
overlaps = 0

lines.each do |line|
    _, id, x, y, w, h = line.match(/#(\d*) @ (\d*),(\d*): (\d*)x(\d*)/).to_a.map(&:to_i)

    for i in x..(x+w-1) do
        for j in y..(y+h-1) do
            coord = i+j*1000
            grid[coord] += 1
            if grid[coord] == 2
                overlaps += 1
            end
        end
    end
end

puts overlaps

File: 2020/03/3_1.py
lines = open('input.txt', 'r').readlines()
map = []
for line in lines:
    map.append(list(line.rstrip()))
    
dx = 3
dy = 1
x = y = 0
height = len(map)
width = len(map[0])
num_trees = 0

while y < height:
    c = map[y][(x % width)]
    if c == '#':
        num_trees += 1
    x += dx
    y += dy
print(num_trees)